### DISCUSSION TITLE
创建自定义聚合函数UDAF时，可变参数用array<double> 报错，请问下目前是不支持么？

### DISCUSSION THREAD

**Discussion by ahboer** [apache/doris#19810] [A - General / Q&A] [answered]
我的UDAF的add方法：
public void add(State state, ArrayList<Double> val)
最初设想是传入不定量的多个字段，结果创建函数时报错：
mysql> CREATE AGGREGATE FUNCTION forcast_flow(array<DOUBLE>) RETURNS array<DOUBLE> PROPERTIES (
    ->     "file"="file:///opt/midian.jar",
    ->     "symbol"="com.my.ForcastUDAF",
    ->     "always_nullable"="true",
    ->     "type"="JAVA_UDF"
    -> );
ERROR 1105 (HY000): errCode = 2, detailMessage = Method 'add' in class 'com.my.ForcastUDAF' does not support non-scalar type 'ARRAY<DOUBLE>'

**Comment by LemonLiTree** [comment]
which doris version？

**Comment by ahboer** [comment]
最新版本1.2.4



***@***.***
 
发件人： Liqf
发送时间： 2023-05-18 19:52
收件人： apache/doris
抄送： ahboer; Author
主题： Re: [apache/doris] 创建自定义聚合函数UDAF时，可变参数用array<double> 报错，请问下目前是不支持么？ (Discussion #19810)
which doris version？
―

**Reply by LemonLiTree** [reply, marked as answer]
Version 1.2 does not support using array in java udf, it will be supported in version 2.0
